packages/wallets/src/features/cfd/modals/MT5PasswordModal/MT5PasswordModal.tsx

```diff
@@ -125,6 +125,7 @@ const MT5PasswordModal: React.FC<TProps> = ({ marketType, platform }) => {
     if (isMobile) {
         return (
             <ModalStepWrapper renderFooter={renderFooter} title={renderTitle()}>
+                {/* TODO: We need to separate this out into a separate `show` modal call to hide the Deriv.app header */}
```

Comment:
There is an issue when dynamically updating the `shouldHideDerivAppHeader` in ModalStepWrapper to hide the Deriv.app header when Success screen is shown.

If we do it like this:
```
<ModalStepWrapper ... shouldHideDerivAppHeader={isSuccess}>
{isSuccess && <Success />}
{!isSuccess && ...}
</ModalStepWrapper>
```
Then when `isSuccess` is set to `true` as soon as the hook returns a successful response, the component will be portal-ed out into the Wallets level root container, BUT the entire component will be re-mounted, and thus `isSuccess` will revert back to `false`.

https://github.com/binary-com/deriv-app/assets/103016120/e0150f6f-f6c4-491d-8179-90c56eb853b2



To fix this, we need to show the Success screen as a separate modal like so (we don't need to pass `shouldHideDerivAppHeader` into the `show` option because we have passed it as a prop to `ModalStepWrapper`:
```
useEffect(() => {
   if (isSuccess) show(
        <ModalStepWrapper .... shouldHideDerivAppHeader>
             <Success .... />
        </ModalStepWrapper />
   )
}, [isSuccess])
```
https://github.com/binary-com/deriv-app/assets/103016120/2f0b3663-e37f-4854-97a7-96b329ffbd48